master/static/migrations/20210203150406_change-limit-behavior.up.sql

```diff
@@ -0,0 +1,24 @@
+CREATE OR REPLACE FUNCTION public.page_info(total bigint, "offset" int, "limit" int)
+    RETURNS json
+    LANGUAGE sql IMMUTABLE
+AS $$
+WITH start_index AS (
+    SELECT (CASE WHEN "offset" < 0
+                 THEN total + "offset"
+                 ELSE "offset" END) AS start_index
+), end_index AS (
+    SELECT (CASE WHEN (SELECT start_index FROM start_index) + "limit" > total THEN total
```

Comment:
I am thinking of this case:
```
offset = a number > total
start_index = offset
end_index = start_index + limit < total
```
We also should fail limit < -2...

Reply:
Ok. -2 already fails and that case (and offset < -total) now just default to returning no rows (which makes plenty of sense to me).